pkg/devcontainer/image_manager.go

```diff
@@ -31,6 +32,15 @@ func NewImageManager(dockerImageCli client.ImageAPIClient, randomString func(int
 func (im *DockerImageManager) PullImage(ctx context.Context, name string) error {
 	log.Debug().Str("image", name).Msg("Pulling image")
 
+	imgs, err := im.ImageList(ctx, image.ListOptions{Filters: filters.NewArgs(filters.Arg("reference", name))})
+	if err != nil {
+		return fmt.Errorf("Failed to list local images, %w", err)
+	}
+	if len(imgs) != 0 {
+		log.Debug().Str("image", name).Msg("Local image exists")
+		return nil
+	}
+
```

Comment:
i think this logic deserves a separate method that we can call from the devcontainer manager and pull the image only if local not found.